website/blog/2024-10-21-creating-an-extension.md

```diff
@@ -0,0 +1,74 @@
+---
+title: How to create your first extension
+description: Learn how to create and customize your own extension for Podman Desktop
+slug: extensions-getting-started
+authors: [cdrage]
+tags: [podman-desktop, release, podman]
+hide_table_of_contents: false
+---
+
+![programming](img/creating-an-extension/programming.png)
+
+# How to create your first extension
+
+Extensions are a powerful tool to customize and extend the functionality of Podman Desktop. Whether you want to add new container management features, streamline current workflows, or create custom UI elements specific to your tech stack, building extensions allows you to tailor the Podman Desktop experience to your specific needs. 
+
+In this guide, we'll introduce how you can build your own Podman Desktop extension, with links to detailed documentation that covers each part of the process.
+
+## Introduction to Extensions
+
+Extensions are abundant in Podman Desktop and can be found in the **Extensions -> Catalog** section.
+
+![extension catalog](img/podman-desktop-release-1.10/extension-catalog.png)
+
+Each extension expands on Podman Desktop, such as providing [Kubernetes development clusters with Minikube](https://github.com/containers/podman-desktop-extension-minikube) or even [analyzing your images](https://github.com/containers/podman-desktop-extension-layers-explorer).
+
+Below we will go over how to start creating your first extension!
+
+## Getting Started with Your Project
+
+The first step in creating your extension is setting up the project environment. To learn how to configure the project and add basic components, check out the **[Templates for creating an extension](/docs/extensions/templates)** guide, which walks you through initializing your project from an official template.
+
+## Adding UI Components
+
+One of the most common tasks when creating an extension is enhancing the user interface. Whether it’s adding buttons, panels, or icons, UI components help make your extension more interactive and accessible. Learn more about this in the **[Adding UI components](/docs/extensions/developing/adding-ui-components)** documentation, where you’ll find instructions on creating and integrating components into the application’s UI.
+
+### Working with Icons
+
+Icons are a great way to make your extension more visually unique. You can learn how to add and style custom icons by following the **[Adding icons](/docs/extensions/developing/adding-icons)** documentation.
```

Comment:
I would add a screenshot of the kind icons to illustrate this concept, what do you think?

Reply:
Added!